gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp

```diff
@@ -197,8 +201,10 @@ void GazeboRosControlPlugin::Load(gazebo::physics::ModelPtr parent, sdf::Element
   rclcpp::Duration gazebo_period(impl_->parent_model_->GetWorld()->Physics()->GetMaxStepSize());
 
   // Decide the plugin control period
-  if (impl_->sdf_->HasElement("control_period")) {
-    impl_->control_period_ = rclcpp::Duration(impl_->sdf_->Get<double>("control_period"));
+  if (sdf->HasElement("control_period")) {
+    impl_->control_period_ = rclcpp::Duration(
```

Comment:
I am not sure if this part needs to change. If control period is in terms of **seconds**, then the following should be fine
```suggestion
    impl_->control_period_ = rclcpp::Duration(sdf->Get<double>("control_period") * 1e9);
```